From d52065b34a66d21410262c38eafdb2a1395120e1 Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Sun, 30 May 2021 12:11:15 +0300
Subject: [PATCH 1570/1921] crypto: inside-secure: support three crypto
 accelerators

increase the number of crypto accelerators supported to three

Signed-off-by: Dana Vardi <danat@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/50327
Reviewed-by: Michael Shamis <michaelsh@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Change-Id: I5690278b5416e43363aae1aed905a1b99b7eff90
Signed-off-by: Ben Peled <bpeled@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52172
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/inside-secure/safexcel.c | 2 +-
 drivers/crypto/inside-secure/safexcel.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index e4ba0b5458ef..5b3436104842 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -28,7 +28,7 @@
 static int eip_in_use = -1;
 
 /* Module param to save the assigned rings to the Kernel */
-static uint rings[MAX_EIP_DEVICE] = {RINGS_UNINITIALIZED, RINGS_UNINITIALIZED};
+static uint rings[MAX_EIP_DEVICE] = {RINGS_UNINITIALIZED, RINGS_UNINITIALIZED, RINGS_UNINITIALIZED};
 
 static void eip197_trc_cache_setupvirt(struct safexcel_crypto_priv *priv)
 {
diff --git a/drivers/crypto/inside-secure/safexcel.h b/drivers/crypto/inside-secure/safexcel.h
index 48f503ad2cd8..df1aef4a9f25 100644
--- a/drivers/crypto/inside-secure/safexcel.h
+++ b/drivers/crypto/inside-secure/safexcel.h
@@ -27,7 +27,7 @@
 						((reg >> 12) & 0xf))
 
 /* Number of eip devices */
-#define MAX_EIP_DEVICE					2
+#define MAX_EIP_DEVICE					3
 
 #define RINGS_UNINITIALIZED				0xff
 
-- 
2.31.1

